right/src/config_parser/parse_config.c

```diff
@@ -76,19 +57,24 @@ parser_error_t ParseConfig(config_buffer_t *buffer)
     uint8_t alphanumericSegmentsBrightness = ReadUInt8(buffer);
     uint8_t keyBacklightBrightness = ReadUInt8(buffer);
 
+    uint32_t ledsFadeTimeout = LedsFadeTimeout;
+    bool perKeyRgbPresent = PerKeyRgbPresent;
+    backlighting_mode_t backlightingMode = BacklightingMode;
+    rgb_t keyActionColors[keyActionColor_Length];
+
     if (DataModelMajorVersion >= 6) {
-        LedsFadeTimeout = 1000 * ReadUInt16(buffer);
-        PerKeyRgbPresent = ReadBool(buffer);
-        BacklightingMode = ReadUInt8(buffer);
-
-        readRgbColor(buffer, KeyActionColor_None);
-        readRgbColor(buffer, KeyActionColor_Scancode);
-        readRgbColor(buffer, KeyActionColor_Modifier);
-        readRgbColor(buffer, KeyActionColor_Shortcut);
-        readRgbColor(buffer, KeyActionColor_SwitchLayer);
-        readRgbColor(buffer, KeyActionColor_SwitchKeymap);
-        readRgbColor(buffer, KeyActionColor_Mouse);
-        readRgbColor(buffer, KeyActionColor_Macro);
+        ledsFadeTimeout = 1000 * ReadUInt16(buffer);
+        perKeyRgbPresent = ReadBool(buffer);
```

Comment:
@kareltucek I am testing the new user-configuration https://github.com/UltimateHackingKeyboard/agent/pull/2166 of the Agent.
I test with the default user configuration. I can save to the keyboard and read from the keyboard. 
After I switch ON the "Per key backlighting" mode I can't save the user config. The firmware returns with 3 - `ParserError_InvalidSerializedKeyActionType`

Could this `perKeyRgbPresent` renaming cause the issue?

Reply:
Well, I dont see anything wrong with this piece of code, but the error is likely caused by something not consuming something that should be consumed.

Reply:
If I read vell the code then 

The appy userfonfiguration process runs first in dry run mode and second in real mode.
The `PerKeyRgbPresent` global variable manages the key actions of the user config contains RGB colors or not.
The default value of the `PerKeyRgbPresent` is false

- the apply user config runs in `dryRun` mode
- here set `perKeyRgbPresent` variable with the userConfig contains RGB colors or not. It is local variable available only in this block
- the `parseKeyActionColor` read the RGB values from the the buffer if `PerKeyRgbPresent` true. In `dryRun` mode the value is always `false` because it set only in real mode. 
- the key action parser reads the first key action from the buffer, but skip the read of RGB data. When the parser reads the 2nd keyaction then the first byte is 255 that is an RGB segment and not a key action type.

I think it is the reason of the error.

Reply:
Yep, that sounds plausible.

Reply:
@ert78gb should be fixed!

Reply:
thx. it works.